From 8653a5cad67abb354cb7ae2cf36b8bc0945baeb1 Mon Sep 17 00:00:00 2001
From: Herrie <Github.com@herrie.org>
Date: Fri, 6 Oct 2023 16:08:52 +0200
Subject: [PATCH] main.qml: Fix the signature of onResponse

Should be using the new signature.

This causes the following in the logs:

2023-10-06T07:40:41.988173Z [5.113598994] daemon.info sh[821] WARNING: file:///usr/lib/maliit/plugins/qml/view-global/main.qml:213:9 Parameter "method" is not declared. Injection of parameters into signal handlers is deprecated. Use JavaScript functions with formal parameters instead.

Signed-off-by: Herman van Hazendonk <github.com@herrie.org>
---
Upstream-Status: Pending

 maliit-plugin-global/view/main.qml | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/maliit-plugin-global/view/main.qml b/maliit-plugin-global/view/main.qml
index 8ba2874..c67066e 100755
--- a/maliit-plugin-global/view/main.qml
+++ b/maliit-plugin-global/view/main.qml
@@ -210,8 +210,12 @@ Rectangle {
         appId: root.appId
         service: "com.webos.settingsservice"
         method: "getSystemSettings"
-        onResponse: {
-            var jsonObject
+        onResponse: (method, payload, token) => {
+            handleResponse(method, payload, token);
+        }
+
+        function handleResponse(method, payload, token) {
+            var jsonObject;
 
             if (method.toString() && method.toString() === "getSystemSettings") {
                 jsonObject = JSON.parse(payload);
